### PR TITLE
Add support for basic authentication with Elasticsearch

### PR DESCRIPTION
**Pull request by mtneug**
Elasticsearch allows for basic authentication. This adds two optional
settings for specifying a username and password. By default these are
null which will deactivate the use of basic authentication (as is the
current behavior).

### Your pull request should…

* [x] have a concise title
* [x] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] be against the correct branch (features can only go into develop)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

### CODE REVIEW

File: modules/search/pom.xml [hunk 1]
```diff
@@ -87,6 +87,15 @@
       <groupId>org.apache.httpcomponents</groupId>
       <artifactId>httpcore-osgi</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.apache.httpcomponents</groupId>
+      <artifactId>httpclient</artifactId>
```
Comment (lkiesow) [inline modules/search/pom.xml]: Isn't all of `httpclient` contained in `httpclient-osgi`?

Comment (mtneug) [reply modules/search/pom.xml]: I have no clue. The dependency analyzer has given me some trouble saying I didn't declare that dependency. The CI build does the same now for the `admin-ui` and I have no idea why 😞 .

**Comment by lkiesow** [comment]
Updated the PR. Can you check this still works for you?

**Comment by gregorydlogan** [comment]
Linking to the relevant docs to enable this on the ES side would be nice, as would adding it to the docker-compose files in the devel-dependency directory if possible.

**Comment by mtneug** [comment]
> Updated the PR. Can you check this still works for you?

Forgot to answer. This is now running and working.

**Comment by mtneug** [comment]
Conflict is fixed

**Comment by mtneug** [comment]
> Linking to the relevant docs to enable this on the ES side would be nice, as would adding it to the docker-compose files in the devel-dependency directory if possible.

I added the reference to the documentation. I wouldn't add it to the developer compose files as these should be kept simple lowering the risk of errors due to misconfiguration.

**Comment by gregorydlogan** [comment]
@mliradelc can we move this forward?

**Comment by lkiesow** [comment]
Since the release is fast approaching, I dare to steal this from you @mliradelc.
We have had two devs from two institutions working on this and reviewing each other.
→ Let's get this in.

**Pull request merged**